src/hotspot/share/gc/serial/cardTableRS.cpp

```diff
@@ -601,20 +345,188 @@ void CardTableRS::verify() {
 CardTableRS::CardTableRS(MemRegion whole_heap) :
   CardTable(whole_heap) { }
 
+// Implemented word-iteration to skip long consecutive clean cards.
+CardTable::CardValue* CardTableRS::find_first_dirty_card(CardValue* const start_card,
+                                                         CardValue* const end_card) {
+  using Word = uintptr_t;
```

Comment:
I think it would be better if Word was consistent with clean_card_row_val, to eliminate casts now and in
the future when you address my earlier comment about needing a clean-row accessor.  So I suggest
`using Word = decltype(clean_card_row_val());`.
Maybe as part of the clean-row accessor also add a CardRow type alias.
Note that I think that type should be unsigned (as currently here, e.g. uintptr_t) rather than signed
(as currently provided by clean_card_row_val).

Reply:
The similar card-scanning code in g1remset also defines `Word`. I will extract them together. This PR targets Serial-specific issue, so I wanna minimize touching the shared code, e.g. introducing a new `CardWord` type and operations around it.